src/fetching/RelayConnectionFetcher.js

```diff
@@ -48,7 +48,7 @@ export default class RelayConnection<T, Data> implements Fetcher<Data[]> {
 		do {
 			const data: GraphQlConnection<T> = await this.fetchConnection();
 
-			yield data.edges.map(edge => this.transformResult(edge.node));
+			yield data.edges.map(edge => this.transformResult(edge.node, data.errors));
```

Comment:
I don't think it makes sense to pass the error to transformResult, this would be called many times when fetching a connection, but it's enough to call it once per page fetch.

Reply:
but each fetch could contain another graphQL error, that transfromResult could handle

Reply:
Yeah, but `transformResult` gets called for each node in the result connection. If you fetch 4 slices of a connection with 10 results each, it would be enough to handle the error 4 times, not 4 * 10 times (which would happen with this solution).

Reply:
right... so mabye we should pass it to the getConnection function then?

Reply:
I don't we should change these functions, I'd prefer a separate optional callback like I mentioned [here](https://github.com/tristramjs/tristram/pull/16#discussion_r218418863)

Reply:
With this implementation, it would be possible to log messages or throw errors if the response includes GraphQL errors. But i think it should be possible to change the transformed output according to the errors.

Reply:
Sounds reasonable. Why not have both? Then we could `transformResults` also based on errors, and separately handle them to e.g. exit early etc.

Reply:
... or we could add a option that a erlay exit could be triggerd with transformResults.
With that an exit could also be triggerd if some imporant graphQL fields are missing (which wont throw any errors)

Reply:
Yup, actually we could do it even without adding another option. Simply throwing in `.map` should exit early.

Reply:
Just leave it like it is right now?

Reply:
atm `transformResults` just gets the node of a connection

Reply:
but in your PR you added `data.errors`, check [l69](https://github.com/tristramjs/tristram/pull/16/files#diff-83053532852658168c27b202bb7c36f3R69)

Reply:
yeah sure, but this conversation is about this line.
So as this conversation still is not resolved, i thougt you meant that this would already be the case whitout the change in this line with this PR 😕

Reply:
Oh, sorry for the confusion!